dotCMS/src/integration-test/java/com/dotmarketing/portlets/workflows/actionlet/SaveContentActionletTest.java

```diff
@@ -83,48 +191,96 @@ private static ContentType createTestType(final ContentTypeAPI contentTypeAPI)
     @AfterClass
     public static void cleanup()
             throws DotDataException, DotSecurityException {
+        for (final ContentType contentType : contentTypes) {
+            if (null != contentType) {
 
-        if (null != customContentType) {
-
-            final ContentTypeAPI contentTypeAPI = APILocator.getContentTypeAPI(APILocator.systemUser());
-            contentTypeAPI.delete(customContentType);
+                final ContentTypeAPI contentTypeAPI = APILocator.getContentTypeAPI(APILocator.systemUser());
+                contentTypeAPI.delete(contentType);
+            }
         }
 
         cleanupDebug(SaveContentActionletTest.class);
     } // cleanup
 
+    /**
+     * Method to test: {@link WorkflowAPI#fireContentWorkflow(Contentlet, ContentletDependencies)}
+     * Given Scenario: Try to Save and publish a {@link Contentlet} with different users, when the ContentType does not
+     * have a 'Site or folder' field
+     * Expected Result: The follow are the expected result according to the user
+     * - If the user is system should save and publish the contentlet
+     * - If the user have {@link PermissionLevel#CAN_ADD_CHILDREN} over the ContentType
+     * - If the user doesn't have any permission should throw a DotSecurityException when try to save it
+     * - If the user just has {@link PermissionLevel#CAN_ADD_CHILDREN} over the save {@link ContentType} then should throw a DotSecurityException when try to save it
+     */
     @Test
-    public void test_Publish_With_Save_Contentlet_Actionlet_Tag () throws DotSecurityException, DotDataException {
+    @UseDataProvider("usersAndContentTypeWithoutHostField")
+    public void test_Publish_With_Save_Contentlet (final TestCase testCase) throws DotSecurityException, DotDataException {
+        final WorkflowAPI workflowAPI = APILocator.getWorkflowAPI();
 
-        final Contentlet contentlet = new Contentlet();
-        contentlet.setContentType(customContentType);
-        contentlet.setProperty("title", "Test");
-        contentlet.setProperty("txt", "Test");
-        contentlet.setProperty("tag", "test");
+        final Contentlet contentlet = new ContentletDataGen(testCase.contentType.id())
+            .setProperty("title", "Test")
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The String literal "Test" appears 6 times in this file; the first occurrence is on line 235](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5337055)